### PR TITLE
[NTOS:SE] Add token debug code

### PR DESCRIPTION
**Pull request by GeoB99**
Implement initial token debug code. For now debug information that is being tracked are: process image file name, process and thread client IDs and token creation method. More specific debug code can be added later only if needed.

As for the token creation method, this follows the same principle as on Windows where the creation method is defined by a value denoting the first letter of the said method of creation. That is, `0xC` is for token creation, `0xD` is for token duplication and `0xF` is for token filtering.

### CODE REVIEW

File: ntoskrnl/se/token.c [hunk 1]
```diff
@@ -1086,6 +1086,17 @@ SepDuplicateToken(
         goto Quit;
     }
 
+    /* Fill in token debug information */
+#if DBG
+    RtlCopyMemory(AccessToken->ProcessName,
+                  PsGetCurrentProcess()->ImageFileName,
+                  16);
```
Comment (binarymaster) [inline ntoskrnl/se/token.c:1093]:
```suggestion
                  sizeof(AccessToken->ProcessName));
```

File: ntoskrnl/include/internal/se.h [hunk 2]
```diff
@@ -34,6 +34,10 @@ typedef struct _TOKEN_AUDIT_POLICY_INFORMATION
     } Policies[1];
 } TOKEN_AUDIT_POLICY_INFORMATION, *PTOKEN_AUDIT_POLICY_INFORMATION;
 
+#define TOKEN_CREATE_METHOD    0xCUL
+#define TOKEN_DUPLICATE_METHOD 0xDUL
+#define TOKEN_FILTER_METHOD    0xFUL
```
Comment (binarymaster) [inline ntoskrnl/include/internal/se.h:39]: CUL DUL FUL! :smile: Sounds nice

File: ntoskrnl/se/token.c [hunk 3]
```diff
@@ -1086,6 +1086,17 @@ SepDuplicateToken(
         goto Quit;
     }
 
+    /* Fill in token debug information */
+#if DBG
+    RtlCopyMemory(AccessToken->ProcessName,
+                  PsGetCurrentProcess()->ImageFileName,
```
Comment (HBelusca) [inline ntoskrnl/se/token.c:1092]: Regarding the bug mentioned on Mattermost, there exists the case, where tokens start to be first used in the kernel, before any process (including the "initial" kernel process) get created; therefore in that case, `PsGetCurrentProcess()` would return `NULL` and you're in trouble.

File: ntoskrnl/se/token.c [hunk 4]
```diff
@@ -1844,6 +1860,43 @@ SepCreateToken(
...
+        /*
+         * Set up the process and thread CIDs.
+         *
+         * !!!IMPORTANT NOTE!!! -- ProcessCid, ThreadCid and CreateMethod
+         * are member field names taken from Windows Server 2003 SP2 checked
+         * build kernel symbols. The reason why we have to have to such names
+         * is for WinDBG debug extension commands support purposes (such as
+         * !logonsession command respectively). THESE FIELD NAMES MUST NOT BE
+         * CHANGED!!!
+         */
```
Comment (learn-more) [inline ntoskrnl/se/token.c:1891]: It makes more sense to have a comment like this in the header where it's defined?

**Pull request merged**